run_integration_tests.go

```diff
@@ -0,0 +1,189 @@
+// +build ignore
+
+package main
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"runtime"
+	"strings"
+)
+
+type CIEnvironment interface {
+	Prepare()
+	RunTests()
+}
+
+type TravisEnvironment struct {
+	goxArch []string
+	goxOS   []string
+}
+
+func (env *TravisEnvironment) Prepare() {
+	msg("preparing environment for Travis CI\n")
+
+	run("go", "get", "github.com/mattn/goveralls")
+	run("go", "get", "github.com/mitchellh/gox")
+
+	if runtime.GOOS == "darwin" {
+		// install the libraries necessary for fuse
+		run("brew", "install", "caskroom/cask/brew-cask")
+		run("brew", "cask", "install", "osxfuse")
+	}
+
+	// only test cross compilation on linux with Travis
+	if runtime.GOOS == "linux" {
+		env.goxArch = []string{"386", "amd64"}
+		if !strings.HasPrefix(runtime.Version(), "go1.3") {
+			env.goxArch = append(env.goxArch, "arm")
+		}
+
+		env.goxOS = []string{"linux", "darwin", "freebsd", "openbsd", "windows"}
+	} else {
+		env.goxArch = []string{runtime.GOARCH}
+		env.goxOS = []string{runtime.GOOS}
+	}
+
+	msg("gox: OS %v, ARCH %v\n", env.goxOS, env.goxArch)
+	run("gox", "-build-toolchain",
+		"-os", strings.Join(env.goxOS, " "),
+		"-arch", strings.Join(env.goxArch, " "))
+}
+
+func (env *TravisEnvironment) RunTests() {
+	// run fuse tests on darwin
+	if runtime.GOOS != "darwin" {
+		msg("skip fuse integration tests on %v\n", runtime.GOOS)
+		os.Setenv("RESTIC_TEST_FUSE", "0")
+	}
+
+	// compile for all target architectures with tags
```

Comment:
I would get rid of all the comments in this function and instead break the code out into smaller functions, i.e. `runFuseTests()`, `compileAllTargetArchitectures()`, `runBuildScript()`, and `gatherCoverageInformation()`. I think that would be more readable.